lib/node_modules/@stdlib/math/generics/statistics/incrprod/test/test.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var incrprod = require( './../lib' );
+var ldexp = require( '@stdlib/math/base/special/ldexp' );
+var EPSILON = require( '@stdlib/math/constants/float64-eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var normal = require( '@stdlib/math/base/random/normal' );
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+    t.ok( true, __filename );
+    t.equal( typeof incrprod, 'function', 'main export is a function' );
+    t.end();
+} );
+
+tape( 'the function returns an accumulator function', function test( t ) {
+    t.equal( typeof incrprod(), 'function', 'returns a function' );
+    t.end();
+} );
+
+tape( 'the accumulator function incrementally computes a product', function test( t ) {
+    var expected;
+    var actual;
+    var data;
+    var acc;
+    var prod;
+    var N;
+    var d;
+    var i;
+
+    data = [ 2.0, 3.0, 2.0, 4.0, 3.0, 4.0 ];
+    N = data.length;
+
+    expected = new Array( N );
+    actual = new Array( N );
+
+    acc = incrprod();
+
+    prod = 1.0;
+    for ( i = 0; i < N; i++ ) {
+        d = data[ i ];
+        prod *= d;
+        expected[ i ] = prod;
+        actual[ i ] = acc( d );
+    }
+
+    t.deepEqual( actual, expected, 'returns expected incremental results' );
+    t.end();
+} );
+
+tape( 'if not provided an input value, the accumulator function returns the current product', function test( t ) {
+    var data;
+    var acc;
+    var i;
+
+    data = [ 2.0, 3.0, 1.0 ];
+    acc = incrprod();
+    for ( i = 0; i < data.length; i++ ) {
+        acc( data[ i ] );
+    }
+    t.equal( acc(), 6.0, 'returns the current accumulated product' );
+    t.end();
+} );
+
+tape( 'calculate the product of a long series defined as a(s) = x if s is even and a(s) = 1/x if s is odd', function test( t ) {
+    var length = 150;
+    var x = 2;
+    var acc;
+    var i;
+
+    acc = incrprod();
+    for( i = 0; i < length; i++ ) {
```

Comment:
Thanks for the tip, I'll check if I can write a short guide from a novice's point of view.
It would be very simple but a base document could be useful regardless.